src/data_gradients/managers/abstract_manager.py

```diff
@@ -63,6 +66,26 @@ def __init__(
         if batches_early_stop:
             logger.info(f"Running with `batches_early_stop={batches_early_stop}`: Only the first {batches_early_stop} batches will be analyzed.")
         self.batches_early_stop = batches_early_stop
+
+        # Check if train_data and val_data are DataLoader objects.
+        # If not, try to convert them to DataLoader objects.
+        # Generally this should work fine, with only caveat that detection datasets may require custom collate_fn.
+        # However since we are using bs=1 this should not be a problem, but just in case we try to get a singe batch
+        # from the dataset and see if it works.
+        if not isinstance(train_data, DataLoader):
+            try:
+                next(iter(DataLoader(train_data)))
```

Comment:
`next(iter(DataLoader(train_data)))` This is to check if we can properly iterate over the DataLoader ?

Reply:
If `train_data` is a generator, this would drop the first element. Maybe it's an edge case that we can live with, but it's not ideal if the user runs on the whole dataset.
The other edge case we discussed was if the user passes an iterable that returns batches but is not a DataLoader.

Do we still want to handle it like that ?
The alternative is to handle the batch/not batch at the batch_processor level, like in segmentation.

Reply:
That's fair. I suggest to ignore this edge case for now 🙈